src/main/java/carpet/mixins/ServerPlayerEntity_scarpetEventMixin.java

```diff
@@ -126,4 +130,16 @@ public boolean isInvalidEntityObject()
     {
         return isInvalidReference;
     }
+
+    //getting player language
+    private String language = null;
```

Comment:
you don't need to == null here (its gonna be added to all constructors if you do it this way)

Reply:
ye I noticed, just wasn't 100% sure when that method got called cos as I mentioned I just coded this up in school without being able to test, so I didn't want NPE